misk-metrics-digester/src/main/kotlin/misk/metrics/digester/Registry.kt

```diff
@@ -0,0 +1,109 @@
+package misk.metrics.digester
+
+import misk.metrics.Histogram
+import misk.metrics.HistogramRecordMetric
+import misk.metrics.HistogramRegistry
+import java.util.Objects
+import java.util.concurrent.ConcurrentHashMap
+
+/** DigestMetric contains the contents of an individual metrics within a histogram */
+class DigestMetric(
+  val digest: SlidingWindowDigest<*>,
+  val labels: List<String>
+) {
+  /** Adds an observation to the metric */
+  fun observe(value: Double) {
+    digest.observe(value)
+  }
+}
+
+class TDigestHistogramRecordMetric constructor(private val metric: DigestMetric) :
+    HistogramRecordMetric {
+  override fun observe(duration: Double) {
+    metric.observe(duration)
+  }
+}
+
+/** TDigestHistogram stores histograms which records metrics */
+class TDigestHistogram<T : TDigest<T>> constructor(
+  val name: String,
+  val help: String,
+  val quantiles: List<Double>,
+  private val labelNames: List<String>,
+  private val tDigest: () -> SlidingWindowDigest<T>
+) : Histogram {
+
+  private val metrics = ConcurrentHashMap<Int, DigestMetric>()
+
+  /** Add all labels passed to the histogram */
+  init {
+    labelNames.forEach { label ->
+      labels(label)
+    }
+  }
+
+  /** Records a new metric within the histogram */
+  override fun labels(vararg labelValues: String): TDigestHistogramRecordMetric {
+    val metric = metrics.getOrPut(key(labelValues)) {
+      DigestMetric(tDigest(), labelValues.asList())
+    }
+    return TDigestHistogramRecordMetric(metric)
+  }
+
+  /** Returns the number of windows within the histogram */
+  override fun count(vararg labelValues: String): Int {
+    return metrics.get(key(labelValues))?.digest?.windows?.count() ?: 0
+  }
+
+  /** Returns a metric of the histogram. Order of labels matters */
+  fun getMetric(vararg metric: String): DigestMetric {
```

Comment:
This doesn't need to be public. Also, Kotlin by convention avoids `get`/`set` prefixes.

```
internal fun metric(vararg metric: String) = metrics[key(metric)]
```

This also changes from `getValue` to normal map get. `getValue` never returns null but your unit tests expect nullable values to come back from this method (`assertThat(histogram.getMetric("label1")).isNotNull()`)